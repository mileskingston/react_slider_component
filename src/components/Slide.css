.slider__slide {
  display: inline-block;
  width: 100%;
  height: 100%;
}
.slider__slide img {
  opacity: 0;
  border: 1px solid green;
}