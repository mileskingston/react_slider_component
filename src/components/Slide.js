import React from 'react';
import PropTypes from 'prop-types';

const Slide = props => (
  <img
    className={`slide ${props.currentSlide ? 'slide--active' : ''}`}
    src={props.url}
    alt={props.alt}
  />
);

Slide.displayName = 'Slide';

Slide.propTypes = {
  url: PropTypes.string.isRequired,
  alt: PropTypes.string,
  currentSlide: PropTypes.bool.isRequired
};

Slide.defaultProps = {
  alt: ''
};

export default Slide;
