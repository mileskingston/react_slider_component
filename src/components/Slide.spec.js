import React from 'react';
import { shallow } from 'enzyme';

import Slide from './Slide';

const mockImage = {
  url: 'test/url',
  alt: 'test alt',
  currentSlide: true
};

describe('Slide component', () => {
    it('should render with matching props', () => {
        const wrapper = shallow(<Slide {...mockImage} />);

        expect(wrapper).toMatchSnapshot();
    });
}); 