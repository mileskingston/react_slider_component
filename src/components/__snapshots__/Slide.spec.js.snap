// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide component should render with matching props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slide
    alt="test alt"
    currentSlide={true}
    url="test/url"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "test alt",
      "className": "slider__slide slider__slide--active",
      "src": "test/url",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "test alt",
        "className": "slider__slide slider__slide--active",
        "src": "test/url",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
